### PR TITLE
Patch for AbstractToken::getIndentation

### PR DESCRIPTION
**Pull request by marcioAlmada**
Solves the negative multiplier warning at str_repeat call (#8)

### CODE REVIEW

**Comment by schmittjoh** [comment]
Could you add a test?

**Comment by marcioAlmada** [comment]
Pushed the tests. You might prefer to assert indentation length, but I don't know exactly how it should behave:
- Does line breaks should count?
- Does non line break white space should count greedily?

I just got rid of the warning message since there is no real need to cirurgically manipulate white space, only preserve original source indentation in some cases.

**Comment by schmittjoh** [comment]
I would prefer to assert the whitespace just like a normal string. 

You could just write a test with the ignore whitespace flag turned off, asserting the current behavior and then ensure that we have the same behavior when the flag is turned on.

**Comment by marcioAlmada** [comment]
Yep, but there is no way to keep the **exact** same behavior; when white space was not ignored and a line started with a blank token `::getIndentation()` was counting its own length as indentation but it should count zero length instead.

Take a look at the new commits above: still backwards compatible and spot only on indentation methods.

**Comment by marcioAlmada** [comment]
In general, the entire project test coverage is very small. I would like to push more automated tests soon.

File: src/JMS/PhpManipulator/TokenStream/AbstractToken.php [hunk 1]
```diff
@@ -215,7 +215,11 @@ public function getIndentation()
     {
         $indentation = $this->getLineIndentation();
 
-        return $indentation.str_repeat(' ', $this->getStartColumn() - strlen($indentation));
+        if($this->isFirstTokenOnLine() AND $this->matches(T_WHITESPACE)) {
```
Comment (schmittjoh) [inline src/JMS/PhpManipulator/TokenStream/AbstractToken.php]: Could you please use `&&`?

File: tests/JMS/PhpManipulator/Tests/TokenStreamTest.php [hunk 2]
```diff
@@ -65,6 +66,39 @@ public function testWhitespace()
...
+    /**
+     * @dataProvider indentationConfigurationProvider
+     */
+    public function testGetIndentation($flag, $lineIndentation, $indentation)
+    {
+        $this->stream->setIgnoreWhitespace($flag);
+        $this->setCode("<?php \n  echo     'foobar';\n        exit;");
+        $i = 0;
+        while ($this->stream->moveNext()) {
+            if( !$this->stream->token instanceof MarkerToken ) {
```
Comment (schmittjoh) [inline tests/JMS/PhpManipulator/Tests/TokenStreamTest.php]: Please add a space after `!` and remove the one before `)`.

**Comment by schmittjoh** [comment]
Looks good otherwise. Regarding test coverage, you're probably right. This was/is mostly tested by another library which depends on it and from which this code was extracted initially.

**Comment by marcioAlmada** [comment]
Ok, Fixed.

> This was/is mostly tested by another library which depends on it and from which this code was extracted initially.

Well, I intend to use this package on a long term project. Maybe I could migrate some tests from that other library you mentioned to this package.

**Pull request merged**